ec/src/models/short_weierstrass/mod.rs

```diff
@@ -105,6 +105,16 @@ pub trait SWCurveConfig: super::CurveConfig {
         res
     }
 
+    /// Default implementation for multi scalar multiplication
+    fn msm(
+        bases: &[Affine<Self>],
+        scalars: &[Self::ScalarField],
+    ) -> Result<Projective<Self>, usize> {
+        (bases.len() == scalars.len())
```

Comment:
IIUC, this is here because calling `VariableBase::msm` results in infinite recursion?

Reply:
Yes, exactly! The previous solution was to redefine in the `VariableBaseMSM`:

```
/// Optimized implementation of multi-scalar multiplication.
    fn msm_bigint(
        bases: &[Self::MulBase],
        bigints: &[<Self::ScalarField as PrimeField>::BigInt],
    ) -> Self {
        unhooked_msm_bigint(based,bigints)
    }

    /// Raw optimized implementation of multi-scalar multiplication, only for usage by curve models.
    fn unhooked_msm_bigint(
        bases: &[Self::MulBase],
        bigints: &[<Self::ScalarField as PrimeField>::BigInt],
    ) -> Self {
        ... current msm_bigint body ...
    }
```

as suggested by @burdges above. But this means to change the `VariableBaseMSM` trait and people looking at this code might not get the reasoning behind it. That's why I favoured re-implement this tiny function body to break the recursive compile time loop.